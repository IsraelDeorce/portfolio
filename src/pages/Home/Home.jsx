import { BlogButton, UIButton } from '../../components'
import { FaGoogleDrive } from 'react-icons/fa'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'

import { TEXT } from '../../utils/constants'

import './Home.scss'

const Placeholder = styled.div`
  scroll-behavior: smooth;
  & > * {
    min-height: 400px;
    background-color: ${({ theme }) => theme.c_importExport};
    border-bottom: 4px solid black;
  }
`

function Home() {
  const { t } = useTranslation()

  return (
    <>
      <section id='home' className='home'>
        <h1>{TEXT.myName}</h1>
        <p className='subtitle'>{t('subtitle')}</p>
        <div className='home-buttons'>
          <BlogButton />
          <UIButton
            onClick={() => window.open('https://docs.google.com/document/d/1gwsxihd7lOVGqM2HzhD2VqOYW85ExDR7hLntG_dOQyg/edit?usp=sharing', '_blank')}
            className='secondary'
            label={t('button.resume')}
            righticon={<FaGoogleDrive />}
          />
        </div>
      </section>
      <Placeholder>
        <section id='sec1' className='sec' />
        <section id='sec2' className='sec' />
        <section id='sec3' className='sec' />
        <section id='sec4' className='sec' />
      </Placeholder>
    </>
  )
}

export default Home