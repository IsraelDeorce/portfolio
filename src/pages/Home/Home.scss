@import '/src/index.scss';

.home {
  background-image: linear-gradient( rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6) ), url('../../assets/Vancouver.jpg');
  background-size: cover;
  background-position: top right 30%;
  background-color: $dark-background;
  color: $dark-brackets;
  padding: 7em 2em 4.5em 2em;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.subtitle {
  font-size: 30px;
}
