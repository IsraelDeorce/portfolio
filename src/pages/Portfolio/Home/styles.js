import styled from 'styled-components'
import Vancouver from '../../../assets/Vancouver.jpg'

export const Home = styled.section`
  background-position: top right 30%;
  background-image: linear-gradient( rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6) ), url(${Vancouver});
  background-size: cover;
  color: #D4D4C8;
  padding-top: 7em;
  padding-bottom: 4.5em;
  align-items: center;
`
export const Title = styled.h1`
  font-family: 'Dancing Script', cursive;
  font-size: 4em;
  font-weight: 400;
  margin: 0;
  margin-bottom: 20px;
`

export const Subtitle = styled.p`
  font-size: 30px;
`
