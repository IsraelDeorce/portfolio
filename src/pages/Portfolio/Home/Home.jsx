import { useTranslation } from 'react-i18next'
import { ExternalButtons } from '../../../components'
import { TEXT } from '../../../utils/constants'
import * as S from './styles'

function Home() {
  const { t } = useTranslation()

  return (
    <S.Home id='home'>
      <h1>{TEXT.MYNAME}</h1>
      <S.Subtitle className='subtitle'>{t('subtitle')}</S.Subtitle>
      <ExternalButtons />
    </S.Home>
  )
}

export default Home
