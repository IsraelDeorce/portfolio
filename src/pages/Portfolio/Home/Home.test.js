import Home from './Home'

jest.mock('react-router-dom')

describe('<Home/>', () => {
  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<Home />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<Home />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<Home />)).toMatchSnapshot()
    })
  })
})
