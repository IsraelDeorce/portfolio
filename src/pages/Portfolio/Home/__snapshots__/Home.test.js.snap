// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Snapshot renders without crashing using mount 1`] = `
<Home>
  <styled.section
    id="home"
  >
    <section
      className="sc-hBMUJo MirVc"
      id="home"
    >
      <h1>
        Israel Deorce
      </h1>
      <styled.p
        className="subtitle"
      >
        <p
          className="sc-fnVZcZ eyxeWs subtitle"
        >
          subtitle
        </p>
      </styled.p>
      <ExternalButtons>
        <styled.div>
          <div
            className="sc-bdnxRM jJHyKE"
          >
            <UIButton
              id="blog-button"
              label="button.visit_my_blog"
            >
              <styled.button
                onClick={[Function]}
                secondary={false}
              >
                <button
                  className="sc-crzoAE hOvMwS"
                  onClick={[Function]}
                >
                  <span>
                    button.visit_my_blog
                  </span>
                </button>
              </styled.button>
            </UIButton>
            <UIButton
              id="resume-button"
              label="button.resume"
              onClick={[Function]}
              rightIcon={<FaGoogleDrive />}
              secondary={true}
            >
              <styled.button
                onClick={[Function]}
                secondary={true}
              >
                <button
                  className="sc-crzoAE foTsck"
                  onClick={[Function]}
                >
                  <span>
                    button.resume
                  </span>
                  <FaGoogleDrive>
                    <IconBase
                      attr={
                        Object {
                          "viewBox": "0 0 512 512",
                        }
                      }
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M339 314.9L175.4 32h161.2l163.6 282.9H339zm-137.5 23.6L120.9 480h310.5L512 338.5H201.5zM154.1 67.4L0 338.5 80.6 480 237 208.8 154.1 67.4z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </FaGoogleDrive>
                </button>
              </styled.button>
            </UIButton>
          </div>
        </styled.div>
      </ExternalButtons>
    </section>
  </styled.section>
</Home>
`;

exports[`<Home/> Snapshot renders without crashing using render 1`] = `
<section
  class="sc-hBMUJo MirVc"
  id="home"
>
  <h1>
    Israel Deorce
  </h1>
  <p
    class="sc-fnVZcZ eyxeWs subtitle"
  >
    subtitle
  </p>
  <div
    class="sc-bdnxRM jJHyKE"
  >
    <button
      class="sc-crzoAE hOvMwS"
    >
      <span>
        button.visit_my_blog
      </span>
    </button>
    <button
      class="sc-crzoAE foTsck"
    >
      <span>
        button.resume
      </span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M339 314.9L175.4 32h161.2l163.6 282.9H339zm-137.5 23.6L120.9 480h310.5L512 338.5H201.5zM154.1 67.4L0 338.5 80.6 480 237 208.8 154.1 67.4z"
        />
      </svg>
    </button>
  </div>
</section>
`;

exports[`<Home/> Snapshot renders without crashing using shallow 1`] = `
<styled.section
  id="home"
>
  <h1>
    Israel Deorce
  </h1>
  <styled.p
    className="subtitle"
  >
    subtitle
  </styled.p>
  <ExternalButtons />
</styled.section>
`;
